File: v1/bench/fib_tail.go
package main

import (
  "fmt"
  "./util"
)

func fib(n, a, b int) int {
  switch n {
  case 0:
    return a
  case 1:
    return b
  default:
    break
  }

  return fib(n-1, b, a+b)
}

func main() {
  fmt.Printf("%v\n", util.Bench(10, func() {
    for i := 0; i < 10000; i++ {
      fib(20, 0, 1)
    }
  }))
}

File: v1/bench/fib_tail.py
from bench import bench

print(bench(10, '''
def fib(n, a, b):
  return a if n == 0 else b if n == 1 else fib(n-1, b, a+b)
''', '''
for _ in range(10000):
  fib(20, 0, 1)
'''))
